File: app/src/main/java/com/huanchengfly/tieba/post/adapters/SearchUserAdapter.java
package com.huanchengfly.tieba.post.adapters;

import android.content.Context;
import android.text.TextUtils;

import com.huanchengfly.tieba.post.api.models.SearchUserBean;
import com.huanchengfly.tieba.post.R;
import com.huanchengfly.tieba.post.utils.ImageUtil;
import com.huanchengfly.tieba.post.utils.NavigationHelper;
import com.huanchengfly.tieba.post.utils.StringUtil;
import com.othershe.baseadapter.ViewHolder;
import com.othershe.baseadapter.base.CommonBaseAdapter;

import java.util.ArrayList;
import java.util.List;

public class SearchUserAdapter extends CommonBaseAdapter<SearchUserBean.UserBean> {
    private NavigationHelper navigationHelper;

    public SearchUserAdapter(Context context) {
        super(context, null, true);
        navigationHelper = NavigationHelper.newInstance(context);
    }

    public void setData(SearchUserBean.SearchUserDataBean data) {
        List<SearchUserBean.UserBean> forumInfoBeans = new ArrayList<>();
        if (data.getExactMatch() != null) forumInfoBeans.add(data.getExactMatch());
        forumInfoBeans.addAll(data.getFuzzyMatch());
        setNewData(forumInfoBeans);
    }

    @Override
    protected void convert(ViewHolder viewHolder, SearchUserBean.UserBean userBean, int position) {
        viewHolder.setText(R.id.item_search_forum_title, StringUtil.getUsernameString(mContext, userBean.getName(), userBean.getUserNickname()));
        viewHolder.setOnClickListener(R.id.item_search_forum, (view) -> {
            navigationHelper.navigationByData(NavigationHelper.ACTION_USER_BY_UID, userBean.getId());
        });
        ImageUtil.load(viewHolder.getView(R.id.item_search_forum_avatar), ImageUtil.LOAD_TYPE_AVATAR, userBean.getPortrait());
        StringBuilder subTitleBuilder = new StringBuilder();
        if (!TextUtils.isEmpty(userBean.getIntro())) {
            subTitleBuilder.append(userBean.getIntro());
            subTitleBuilder.append("\n");
        }
        subTitleBuilder.append(mContext.getString(R.string.fans_num, userBean.getFansNum()));
        viewHolder.setText(R.id.item_search_forum_subtitle, subTitleBuilder.toString());
    }

    @Override
    protected int getItemLayoutId() {
        return R.layout.item_search_user;
    }
}

File: app/src/main/java/com/huanchengfly/tieba/post/adapters/SearchUserAdapter.kt
package com.huanchengfly.tieba.post.adapters

import android.content.Context
import android.text.TextUtils
import com.alibaba.android.vlayout.layout.LinearLayoutHelper
import com.huanchengfly.tieba.post.R
import com.huanchengfly.tieba.post.adapters.base.BaseSingleTypeDelegateAdapter
import com.huanchengfly.tieba.post.api.models.SearchUserBean
import com.huanchengfly.tieba.post.components.MyViewHolder
import com.huanchengfly.tieba.post.utils.ImageUtil
import com.huanchengfly.tieba.post.utils.StringUtil
import com.huanchengfly.tieba.post.utils.getIntermixedColorBackground

class SearchUserAdapter(
        context: Context
) : BaseSingleTypeDelegateAdapter<SearchUserBean.UserBean>(
        context, LinearLayoutHelper()
) {
    override fun convert(viewHolder: MyViewHolder, item: SearchUserBean.UserBean, position: Int) {
        viewHolder.setText(R.id.item_search_user_title, StringUtil.getUsernameString(context, item.name, item.userNickname))
        ImageUtil.load(viewHolder.getView(R.id.item_search_user_avatar), ImageUtil.LOAD_TYPE_AVATAR, item.portrait)
        val subTitleBuilder = StringBuilder()
        if (!TextUtils.isEmpty(item.intro)) {
            subTitleBuilder.append(item.intro)
            subTitleBuilder.append("\n")
        }
        subTitleBuilder.append(context.getString(R.string.fans_num, item.fansNum))
        viewHolder.setText(R.id.item_search_user_subtitle, subTitleBuilder.toString())
        viewHolder.itemView.background = getIntermixedColorBackground(
                context,
                position,
                itemCount,
                positionOffset = 1,
                colors = intArrayOf(R.color.default_color_card, R.color.default_color_divider),
                radius = context.resources.getDimension(R.dimen.card_radius)
        )
    }

    override fun getItemLayoutId(): Int {
        return R.layout.item_search_user
    }
}